client/engine/engine.go

```diff
@@ -195,12 +194,11 @@ func (e *Engine) handleProposal(proposal consensus_channel.Proposal) (EngineEven
 func (e *Engine) handleMessage(message protocols.Message) (EngineEvent, error) {
```

Comment:
Outside the scope of this PR; it still seems a little unclear exactly what messages we are supporting. This PR cleans up what an objective payload can be, but it doesn't yet help us understand whether a `protocols.Message` could contain e.g. several objective payloads, several payments, several rejected objectives, several ledger proposals all at the same time. Could we one day we can return to this and do a similar cleanup -- i.e. have a top level `MessageType`, serialized message body (bytes) and a finite number of concrete types which that body can be serialized into? Thoughts?

Reply:
This is a good point! Right now our message format is designed to contain multiple "things", however most times when we send a message I think it generally contains one "thing".

One approach is follow the `ObjectivePayload`. A message could have some kind of type that we switch on and then we decode the data based on the type. A message would only ever be one "thing".

Is there any advantage of having a message format that could contain multiple payloads/payments/etc.. (like we do now)? Right now I don't think there's any case where a protocol is sending out multiple messages which could benefit from consolidating into one message.